model/src/gateway/event/shard.rs

```diff
@@ -85,3 +87,21 @@ pub enum ShardEvent {
     /// A shard is now in a Resuming stage after a disconnect.
     Resuming(Resuming),
 }
+
+impl TryFrom<Event> for ShardEvent {
+    type Error = &'static str;
+
+    fn try_from(event: Event) -> Result<Self, Self::Error> {
+        match event {
+            Event::ShardConnected(v) => Ok(Self::Connected(v)),
+            Event::ShardConnecting(v) => Ok(Self::Connecting(v)),
+            Event::ShardDisconnected(v) => Ok(Self::Disconnected(v)),
+            Event::ShardIdentifying(v) => Ok(Self::Identifying(v)),
+            Event::ShardPayload(v) => Ok(Self::Payload(v)),
+            Event::ShardReconnecting(v) => Ok(Self::Reconnecting(v)),
+            Event::ShardResuming(v) => Ok(Self::Resuming(v)),
+
+            _ => Err("event was not a ShardEvent"),
```

Comment:
use "is" instead of "was" to make consistent with other errors in this changeset